src/lib/accounts/AccountSuperhero.ts

```diff
@@ -34,84 +31,47 @@ export class AccountSuperhero extends AccountBase {
     this.address = getLastActiveProtocolAccount(PROTOCOL_AETERNITY)!.address;
   }
 
-  signTransaction(txBase64: Encoded.Transaction, options: any): Promise<Encoded.Transaction> {
-    if (IS_MOBILE_APP && options.aeppOrigin) {
-      return this.fgPermissionCheckAndSign(POPUP_TYPE_SIGN, txBase64, options, options.aeppOrigin);
-    }
-    if (IS_EXTENSION_BACKGROUND) {
-      return this.bgPermissionCheckAndSign(
-        POPUP_TYPE_TX_SIGN,
-        txBase64,
-        { ...options, origin: options.aeppOrigin },
-      );
-    }
-    return this.store.dispatch('accounts/signTransaction', {
-      txBase64,
-      options: {
-        fromAccount: this.address,
-        ...options,
-      },
-    });
-  }
+  async signTransaction(
+    txBase64: Encoded.Transaction,
+    options: Parameters<AccountBase['signTransaction']>[1],
+  ): Promise<Encoded.Transaction> {
+    const { aeppOrigin } = options;
+    const app = aeppOrigin ? new URL(aeppOrigin) : null;
+    const { checkPermission } = usePermissions();
 
-  async signMessage(message: string, options: any): Promise<Uint8Array> {
-    if ((IS_MOBILE_APP || IN_FRAME) && options.aeppOrigin) {
-      return this.fgPermissionCheckAndSign('message.sign', message, options, options.aeppOrigin);
-    }
-    if (IS_EXTENSION_BACKGROUND) {
-      return this.bgPermissionCheckAndSign(
-        METHODS.signMessage,
-        message,
-        { ...options, origin: options.aeppOrigin },
-      );
-    }
-    return this.sign(messageToHash(message), { fromAccount: this.address, ...options });
-  }
-
-  sign(data: string | Uint8Array, options: any): Promise<Uint8Array> {
-    const { getLastActiveProtocolAccount } = useAccounts();
-    return IS_EXTENSION_BACKGROUND
-      ? sign(data, getLastActiveProtocolAccount(PROTOCOL_AETERNITY)!.secretKey) as any
-      : this.store.dispatch('accounts/sign', {
-        data,
-        options: {
-          fromAccount: this.address,
-          ...options,
-        },
-      });
-  }
-
-  async fgPermissionCheckAndSign(method: any, payload: any, options: any, origin?: string) {
     try {
-      const app = origin ? new URL(origin) : null;
-      const permission = (!origin && IS_MOBILE_APP) || await this.store.dispatch('permissions/checkPermissions', {
-        host: app ? app.host : null,
-        method,
-        params: payload,
-      });
-      if (method === 'message.sign') {
-        if (!permission) {
-          const { openModal } = useModals();
-          await openModal(MODAL_MESSAGE_SIGN, {
-            message: payload,
-            app: {
-              name: app?.hostname,
-              icons: [],
-              protocol: app?.protocol,
-              host: app?.host,
-              url: app?.href,
+      if (IS_EXTENSION_BACKGROUND && aeppOrigin) {
+        const txObject = unpackTx(txBase64);
+        if (
+          (checkPermission(new URL(aeppOrigin).host, METHODS.sign, txObject as any))
```

Comment:
Why `as any` and not `ITransactionSignPermissionOptions`

Reply:
This was a temporary thing. I did some research and the `txObject` has `amount` property which fits the requirement of the `checkPermission` method. So I changed the `ITransactionSignPermissionOptions` type to allow to pass any object that has at least one of the required properties: `amount`, `fee` or `nameFee`. Now there is no need to use `as` here.